src/main/java/com/celements/pagelayout/DefaultLayoutService.java

```diff
@@ -0,0 +1,491 @@
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xpn.xwiki.plugin.packaging.DocumentInfo;
+import com.xpn.xwiki.plugin.packaging.PackageAPI;
+
+@Component
+@ThreadSafe
+public final class DefaultLayoutService implements LayoutServiceRole {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DefaultLayoutService.class);
+
+  private static final String CEL_RENDERING_LAYOUT_STACK_PROPERTY = "celRenderingLayoutStack";
+  private static final String CEL_RENDERING_LAYOUT_CONTEXT_PROPERTY = "celRenderingLayout";
+
+  static final String SIMPLE_LAYOUT = "SimpleLayout";
+  static final String XWIKICFG_CELEMENTS_LAYOUT_DEFAULT = "celements.layout.default";
+
+  /**
+   * The name of the internal packaging plugin.
+   */
+  private static final String PACKAGEPLUGIN_NAME = "package";
+
+  @Requirement
+  private Execution execution;
+
+  @Requirement
+  private ModelContext modelContext;
+
+  @Requirement
+  private ModelUtils modelUtils;
+
+  @Requirement
+  private IModelAccessFacade modelAccess;
+
+  @Requirement
+  QueryManager queryManager;
+
+  @Requirement
+  ConfigurationSource cfgSrc;
+
+  @Override
+  public final Map<SpaceReference, String> getAllPageLayouts() {
+    return getPageLayoutMap(false);
+  }
+
+  @Override
+  public final Map<SpaceReference, String> getActivePageLayouts() {
+    return getPageLayoutMap(true);
+  }
+
+  private ImmutableMap<SpaceReference, String> getPageLayoutMap(boolean onlyActive) {
+    Builder<SpaceReference, String> plMapBuilder = ImmutableMap.<SpaceReference, String>builder();
+    try {
+      Query theQuery = queryManager.createQuery(getPageLayoutHQL(onlyActive), Query.HQL);
+
+      List<String[]> results = theQuery.execute();
+      for (String[] resultRow : results) {
+        plMapBuilder.put(RefBuilder.create()
+            .with(modelContext.getWikiRef())
+            .space(resultRow[0])
+            .build(SpaceReference.class),
+            Strings.nullToEmpty(resultRow[1]));
+
+      }
+    } catch (QueryException exp) {
+      LOGGER.error("Failed to get all page layouts", exp);
+    }
+    return plMapBuilder.build();
+  }
+
+  String getPageLayoutHQL(boolean onlyActive) {
+    String hql = "select doc.space, pl.prettyname"
+        + " from XWikiDocument as doc, BaseObject obj, Celements.PageLayoutPropertiesClass as pl"
+        + " where doc.fullName = obj.name"
+        + " and obj.className='Celements.PageLayoutPropertiesClass'"
+        + " and pl.id.id=obj.id";
+    if (onlyActive) {
+      hql += " and pl.isActive = 1";
+    }
+    hql += " order by pl.prettyname asc";
+    return hql;
+  }
+
+  @Override
+  public final boolean createLayout(SpaceReference layoutSpaceRef) {
+    if (layoutSpaceRef != null) {
+      if (!existsLayout(layoutSpaceRef)) {
+        try {
+          XWikiDocument propXdoc = modelAccess
+              .getOrCreateDocument(getLayoutPropDocRef(layoutSpaceRef).get());

```

Comment:
Avoid `get` without preceding `isPresent` check or:
```suggestion
          XWikiDocument propXdoc = getLayoutPropDocRef(layoutSpaceRef)
              .map(modelAccess::getOrCreateDocument)
              .orElseThrow(IllegalArgumentException::new);
```